### PR TITLE
data-types-overview tasks are completed

### PR DESCRIPTION
**Pull request by emakhachek**

### CODE REVIEW

File: Exercises/data-types-overview/data-types-overview.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+console.log(typeof e); // expected output: "symbol"
+let f = null;
+console.log(typeof f); // expected output (well-known js bug): object
+let g;
+console.log(typeof g); // expected output: undefined
+let h = 1n;
+console.log(typeof h); // expected output: bigint
+
+// Task #3
+let objectDemonstration = {
```
Comment (lizarusi) [inline Exercises/data-types-overview/data-types-overview.js]: by `reference` meant that you need to show that variable is just `reference` to object

File: Exercises/data-types-overview/data-types-overview.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+// Task #4
+let a = 2;
+console.log(`результат: ${a * 5}`); // результат: 10
+
+// Additional task #1
+let testArray = ['String', { objectKey: 'value' }, true, 58, null, Symbol("id"), undefined, 58n];
+
+// Additional task #2
+let result = {};
```
Comment (lizarusi) [inline Exercises/data-types-overview/data-types-overview.js]: use `const` here

File: Exercises/data-types-overview/data-types-overview.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+for (let i = 0; i < testArray.length; i++) {
+    let currentType = typeof testArray[i];
```
Comment (lizarusi) [inline Exercises/data-types-overview/data-types-overview.js]: `const`  here as well

File: Exercises/data-types-overview/data-types-overview.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+alert("fullName" in objectDemonstration); //false
+alert(a in objectDemonstration); //true
+
+
+// Task #4
+let a = 2;
+console.log(`результат: ${a * 5}`); // результат: 10
+
+// Additional task #1
+let testArray = ['String', { objectKey: 'value' }, true, 58, null, Symbol("id"), undefined, 58n];
```
Comment (lizarusi) [inline Exercises/data-types-overview/data-types-overview.js]: always `const` when you are not going to reassign

**Review by lizarusi — CHANGES_REQUESTED** [review]
Great job! Small changes required

File: Exercises/data-types-overview/data-types-overview.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+let bString = new String('abc');
+console.log(typeof bString); // expected output: object
+let c = true;
+console.log(typeof c); // expected output: boolean
+let cBoolean = new Boolean(true);
+console.log(typeof cBoolean); // expected output: object
+let d = { da: 1, db: 2 };
+console.log(typeof d); // expected output: object
+let dObject = new Object('{ da: 1, db: 2 }');
+console.log(typeof dObject); // expected output: object
```
Comment (Mike-Savin) [inline Exercises/data-types-overview/data-types-overview.js:17]: Not an object but a new string will be created:

image

File: Exercises/data-types-overview/data-types-overview.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+// Additional task #1
+const testArray = ['String', { objectKey: 'value' }, true, 58, null, Symbol("id"), undefined, 58n];
+
+// Additional task #2
+const result = {};
+for (let i = 0; i < testArray.length; i++) {
+    const currentType = typeof testArray[i];
+    if (result[currentType] === undefined) {
+        result[currentType] = 1
```
Comment (Mike-Savin) [inline Exercises/data-types-overview/data-types-overview.js]: It's better not to miss the semicolon.

**Review by Mike-Savin — CHANGES_REQUESTED** [review]

File: Exercises/data-types-overview/data-types-overview.js (hunk 5 above)
Comment (Mike-Savin) [reply Exercises/data-types-overview/data-types-overview.js:17]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object

Comment (emakhachek) [reply Exercises/data-types-overview/data-types-overview.js:17]: Disscussed in person: decided to leave as it is.
console.log(typeof dObject); shows Object, but dObject is a String.

**Review by Mike-Savin — APPROVED** [review]
👍

**Review by lizarusi — APPROVED** [review]

**Pull request merged**